Source/SwiftLintBuiltInRules/Rules/Style/RedundantDiscardableLetRule.swift

```diff
@@ -32,10 +51,87 @@ struct RedundantDiscardableLetRule: Rule {
                 }
                 """),
+            Example("""
+                @ViewBuilder
+                func bar() -> some View {
+                    ↓let _ = foo()
+                    return Text("Hello, World!")
+                }
+                """),
+            Example("""
+                #Preview {
+                    ↓let _ = foo()
+                    return Text("Hello, World!")
+                }
+                """),
+            Example("""
+                static var previews: some View {
+                    ↓let _ = foo()
+                    Text("Hello, World!")
+                }
+                """),
+            Example("""
+                var notBody: some View {
+                    ↓let _ = foo()
+                    Text("Hello, World!")
+                }
+                """, configuration: ["ignore_swiftui_view_bodies": true], excludeFromDocumentation: true),
+            Example("""
+                var body: some NotView {
+                    ↓let _ = foo()
+                    Text("Hello, World!")
+                }
+                """, configuration: ["ignore_swiftui_view_bodies": true], excludeFromDocumentation: true),
         ],
         corrections: [
             Example("↓let _ = foo()"): Example("_ = foo()"),
             Example("if _ = foo() { ↓let _ = bar() }"): Example("if _ = foo() { _ = bar() }"),
+            Example("""
+                var body: some View {
+                    ↓let _ = foo()
+                    Text("Hello, World!")
+                }
+                """): Example("""
+                    var body: some View {
+                        _ = foo()
+                        Text("Hello, World!")
+                    }
+                    """),
+            Example("""
+                @ViewBuilder
+                func bar() -> some View {
+                    ↓let _ = foo()
+                    return Text("Hello, World!")
+                }
+                """): Example("""
+                    @ViewBuilder
+                    func bar() -> some View {
+                        _ = foo()
+                        return Text("Hello, World!")
+                    }
+                    """),
+            Example("""
+                #Preview {
+                    ↓let _ = foo()
+                    return Text("Hello, World!")
+                }
+                """): Example("""
+                    #Preview {
+                        _ = foo()
+                        return Text("Hello, World!")
+                    }
+                    """),
+            Example("""
+                static var previews: some View {
+                    ↓let _ = foo()
+                    Text("Hello, World!")
+                }
+                """): Example("""
+                    static var previews: some View {
+                        _ = foo()
+                        Text("Hello, World!")
+                    }
+                    """),
```

Comment:
I think that two examples are enough to show that corrections work. Please consider adding another example with the option enabled to show that no correction would be applied.